File: PODM/SW/pod-manager/podm-business-entities/src/main/java/com/intel/podm/business/entities/redfish/properties/MemoryLocation.java
/*
 * Copyright (c) 2016 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.intel.podm.business.entities.redfish.properties;

import com.intel.podm.business.entities.base.DomainObject;
import com.intel.podm.business.entities.base.DomainObjectProperty;

import javax.enterprise.context.Dependent;
import javax.transaction.Transactional;

import static com.intel.podm.business.entities.base.DomainObjectProperties.integerProperty;
import static javax.transaction.Transactional.TxType.MANDATORY;

@Dependent
@Transactional(MANDATORY)
public class MemoryLocation extends DomainObject {
    public static final DomainObjectProperty<Integer> SOCKET = integerProperty("socket");
    public static final DomainObjectProperty<Integer> MEMORY_CONTROLLER = integerProperty("memoryController");
    public static final DomainObjectProperty<Integer> CHANNEL = integerProperty("channel");
    public static final DomainObjectProperty<Integer> SLOT = integerProperty("slot");

    public Integer getSocket() {
        return getProperty(SOCKET);
    }

    public void setSocket(Integer locationSocket) {
        setProperty(SOCKET, locationSocket);
    }

    public Integer getMemoryController() {
        return getProperty(MEMORY_CONTROLLER);
    }

    public void setMemoryController(Integer locationMemoryController) {
        setProperty(MEMORY_CONTROLLER, locationMemoryController);
    }

    public Integer getChannel() {
        return getProperty(CHANNEL);
    }

    public void setChannel(Integer locationChannel) {
        setProperty(CHANNEL, locationChannel);
    }

    public Integer getSlot() {
        return getProperty(SLOT);
    }

    public void setSlot(Integer locationSlot) {
        setProperty(SLOT, locationSlot);
    }
}

File: PSME/common/agent-framework/src/module/model/attributes/memory_location.cpp
/*!
 * @section LICENSE
 *
 * @copyright
 * Copyright (c) 2015-2019 Intel Corporation
 *
 * @copyright
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * @copyright
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * @copyright
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * @section DESCRIPTION
 *
 * */

#include "agent-framework/module/model/attributes/memory_location.hpp"
#include "agent-framework/module/constants/compute.hpp"

using namespace agent_framework::model::attribute;
using namespace agent_framework::model;

MemoryLocation::MemoryLocation() { }

MemoryLocation::~MemoryLocation() { }

json::Json MemoryLocation::to_json() const {
    json::Json result = json::Json();
    result[literals::MemoryLocation::SOCKET] = get_socket();
    result[literals::MemoryLocation::CONTROLLER] = get_controller();
    result[literals::MemoryLocation::CHANNEL] = get_channel();
    result[literals::MemoryLocation::SLOT] = get_slot();
    return result;
}

MemoryLocation MemoryLocation::from_json(const json::Json& json) {
    attribute::MemoryLocation location{};
    location.set_socket(json[literals::MemoryLocation::SOCKET]);
    location.set_controller(json[literals::MemoryLocation::CONTROLLER]);
    location.set_channel(json[literals::MemoryLocation::CHANNEL]);
    location.set_slot(json[literals::MemoryLocation::SLOT]);
    return location;
}
